---
title: Galaxy workflows for SARS-CoV-2 data analysis
---


## An introduction to Galaxy
[A Short Introduction to Galaxy](https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-short/slides-plain.html)

## Galaxy worflows for SARS-CoV-2 data analysis

We will demonstrate how to analyze SARS-COV-2 ampliconic data generated by ARTIC
protocols and sequenced on Illumina and Oxford Nanopore machines.
For the analysis, we will make use of the tutorial developed by galaxy core members:

[Mutation calling, viral genome reconstruction and lineage/clade assignment from SARS-CoV-2 sequencing data](https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/sars-cov-2-variant-discovery/tutorial.html)



## Oxford Nanopore ARTIC data analysis with Galaxy
For analysis of ampliconic ARTIC data, we will make use of the following datasets
            
## Data
```
https://hpc.ilri.cgiar.org/~gkibet/SARS-ONT/fastq/barcode85.fastq.gz
https://hpc.ilri.cgiar.org/~gkibet/SARS-ONT/fastq/barcode86.fastq.gz
https://hpc.ilri.cgiar.org/~gkibet/SARS-ONT/fastq/barcode87.fastq.gz
https://hpc.ilri.cgiar.org/~gkibet/SARS-ONT/fastq/barcode88.fastq.gz
https://hpc.ilri.cgiar.org/~gkibet/SARS-ONT/fastq/barcode89.fastq.gz
```

### Find data on SRA
First we need to find a good dataset to play with. The [Sequence Read Archive (SRA)](https://www.ncbi.nlm.nih.gov/sra) is the primary archive of unassembled reads operated by the US [National Institutes of Health (NIH)](https://www.ncbi.nlm.nih.gov/). SRA is a great place to get the sequencing data that underlie publications and studies. Let’s do that:
            
1. Go to NCBI’s SRA page by pointing your browser to [SRA](https://www.ncbi.nlm.nih.gov/sra)
2. In the search box enter PRJNA909758 and click Search button. (Alternatively, you simply click on this [link](https://www.ncbi.nlm.nih.gov/sra/?term=PRJNA909758))
3. Download metadata describing these datasets by:

    - clicking on **Send** to: dropdown
    - Selecting **File**
    - Changing Format to **RunInfo**
    - Clicking **Create file**
4. This would create a rather large `SraRunInfo.csv` file in your `Downloads` folder.

Now that we have downloaded this file we can go to a Galaxy instance and start processing it.
            
We will follow these steps, highlighted in [From NCBI's Sequence Read Archive (SRA) to Galaxy: SARS-CoV-2 variant analysis](https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html) to get a subset of the data:
    
1. Process and filter ```SraRunInfo.csv``` file in Galaxy
2. Here the `sra accessions` we will use are: `SRR22561688` and `SRR22561690`
3. Download sequencing data with ```Faster Download and Extract Reads in FASTQ```
            
            

## Activity            
            
## Galaxy Exercise using MiSeq Data            
This activity it to help you practice the analysis on illumina data using the [Galaxy workflow](https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/sars-cov-2-variant-discovery/tutorial.html
).      
You are encouraged to change the parameters including: sample IDs (names), coverage depth cutoffs,...            
Follow the steps as guided we used in the analysis of Illumina sequenced data  
### Data
```
https://hpc.ilri.cgiar.org/~jjuma/SARS-CoV-2-ILRI-WHO-2022/testdata-subsampled/COVM02379_S37_con_R1_001.fastq.gz
https://hpc.ilri.cgiar.org/~jjuma/SARS-CoV-2-ILRI-WHO-2022/testdata-subsampled/COVM02379_S37_con_R2_001.fastq.gz
https://hpc.ilri.cgiar.org/~jjuma/SARS-CoV-2-ILRI-WHO-2022/testdata-subsampled/COVM02720_S365_con_R1_001.fastq.gz
https://hpc.ilri.cgiar.org/~jjuma/SARS-CoV-2-ILRI-WHO-2022/testdata-subsampled/COVM02720_S365_con_R2_001.fastq.gz          
```            

### Additional Illumina data.
The data below comes from a MiSeq sequencer            
```
https://hpc.ilri.cgiar.org/~gkibet/SARS-ILL/fastq/sample1_S2_L001_R1_001.fastq.gz
https://hpc.ilri.cgiar.org/~gkibet/SARS-ILL/fastq/sample1_S2_L001_R2_001.fastq.gz
https://hpc.ilri.cgiar.org/~gkibet/SARS-ILL/fastq/sample2_S1_L001_R1_001.fastq.gz
https://hpc.ilri.cgiar.org/~gkibet/SARS-ILL/fastq/sample2_S1_L001_R2_001.fastq.gz
https://hpc.ilri.cgiar.org/~gkibet/SARS-ILL/fastq/sample3_S3_L001_R1_001.fastq.gz
https://hpc.ilri.cgiar.org/~gkibet/SARS-ILL/fastq/sample3_S3_L001_R2_001.fastq.gz
https://hpc.ilri.cgiar.org/~gkibet/SARS-ILL/fastq/sample4_S6_L001_R1_001.fastq.gz
https://hpc.ilri.cgiar.org/~gkibet/SARS-ILL/fastq/sample4_S6_L001_R2_001.fastq.gz
https://hpc.ilri.cgiar.org/~gkibet/SARS-ILL/fastq/sample5_S8_L001_R1_001.fastq.gz
https://hpc.ilri.cgiar.org/~gkibet/SARS-ILL/fastq/sample5_S8_L001_R2_001.fastq.gz            
```            

<br>